include/swift/Basic/type_traits.h

```diff
@@ -29,7 +29,7 @@ namespace swift {
 /// is not intended to be specialized.
 template<typename T>
 struct IsTriviallyCopyable {
-#if _LIBCPP_VERSION
+#if _LIBCPP_VERSION || defined(_MSC_VER)
```

Comment:
Right, that was my concern. Would `|| (defined(_MSC_VER) && !defined(__clang__)` work? I dunno how to detect clang-cl!

Maybe its a version thing - we should check `_MSC_VER >= X`?